agent/cache-types/streaming_health_services_test.go

```diff
@@ -43,7 +43,7 @@ func TestStreamingHealthServices_EmptySnapshot(t *testing.T) {
 	req := &structs.ServiceSpecificRequest{
 		Datacenter:     "dc1",
 		ServiceName:    "web",
-		EnterpriseMeta: structs.EnterpriseMetaInitializer(namespace),
+		EnterpriseMeta: structs.NewEnterpriseMeta(namespace),
```

Comment:
Just curious but is this still idiomatic naming if the new value isn't heap allocated. The reason I didn't go with `New*` originally was so that a caller could know by the name alone that this isn't doing an allocation.

Reply:
Good question! I have never heard of `New` functions only being used when a pointers or interfaces is returned.

I suspect that using `func New` for non-pointer returns is rare because generally structs in Go are either for data (in which case the zero value should be usable and there is no need for a constructor), or they are used for behaviour with pointer receiver methods (and a `func New` is necessary for creating one).

I think our use here is unique because normally we wouldn't need a `func New` normally, we only need it for enterprise differentiation.

I couldn't find any instances in the stdlib (likely because they would not use a constructor for this type). CodeReviewComments includes at least this one example: https://github.com/golang/go/wiki/CodeReviewComments#interfaces showing a `func New` that returns a struct (not a pointer).

Reply:
My understanding is that the compiler could still decide to heap allocate this variable if the code takes the address to it later on. The programmer doesn't necessary get to decide if it is heap or stack allocated, so most likely naming things with that assumption won't work out well.